java/ql/src/Likely Bugs/Concurrency/DoubleCheckedLockingShared.qhelp

```diff
@@ -66,6 +66,26 @@ variable can be used to avoid reading the field more times than neccessary.
 </p>
 <sample src="DoubleCheckedLockingGood.java"/>
 
+<p>
+As a final note, it is possible to use double-checked locking correctly without
+<code>volatile</code> if the constructed object is immutable in the sense that
+all its fields are declared <code>final</code> and the double-checked field is
+read exactly once outside the synchronized block.
+</p>
```

Comment:
Suggest splitting into two sentences:

As a final note, it is possible to use double-checked locking correctly without
<code>volatile</code> if the object you construct is immutable. That is, the object declares all fields as <code>final</code>, and the double-checked field is read exactly once outside the synchronized block.

Reply:
I just updated the qhelp to this, but then noticed that the result is wrong, since the "That is,.." clarification only applies to "immutable", and the "and the double-checked field is read exactly once.." is part of the previous sentence as part of the condition on correctness.  So I'll try with a parenthesis instead even though that still becomes a rather long sentence.

Reply:
My apologies for the poor suggestion. I think that the parentheses are helpful.